### PR TITLE
Datetime fix to standard messaging style (Ex. "Today", "Yesterday").  Additional capability to show display name of user that sent message.  Additional capability to only show display name and avatar for first message from user.

### PR DESCRIPTION
**Pull request by blakesplay**
A couple of changes that add some features that group messaging apps usually have.  I will add options later to inits as per last time.

### CODE REVIEW

**Comment by EnesKaraosman** [comment]
Thanks for PR @blakesplay , I'll review when I'm available

File: Sources/SwiftyChat/ChatView.swift [hunk 1]
```diff
@@ -114,7 +121,7 @@ public struct ChatView<Message: ChatMessage, User: ChatUser>: View {
         )
         .onTapGesture { onMessageCellTapped(message) }
         .contextMenu(menuItems: { messageCellContextMenu(message) })
-        .modifier(AvatarModifier<Message, User>(message: message))
+        .modifier(AvatarModifier<Message, User>(message: message, show: (avatarShow || !shouldShowGroupChatHeaders)))
```
Comment (gabrielPeart) [inline Sources/SwiftyChat/ChatView.swift:124]: Maybe here we could add a method that returns a Boolean value. Give it a name that illustrates the idea and abstract the logical evaluation to this method

File: Sources/SwiftyChat/ChatView.swift [hunk 2]
```diff
@@ -139,6 +146,31 @@ public extension ChatView {
...
+            message.id == thisMessage.id
+        }
+        if(messageIndex == 0){
+            return true
+        }else if(messageIndex == nil){
+            return false
+        }else{
+            let prevMessageUserID = messages[messageIndex!].user.id
+            let currMessageUserID = messages[messageIndex! - 1].user.id
+            if(prevMessageUserID == currMessageUserID){
```
Comment (gabrielPeart) [inline Sources/SwiftyChat/ChatView.swift:166]: Maybe here we could just 

return prevMessagsUserID != currentMessageUserID

And skip the IF/ELSE part?

File: Sources/SwiftyChat/Common/Modifiers/AvatarModifier.swift [hunk 3]
```diff
@@ -65,18 +67,25 @@ internal struct AvatarModifier<Message: ChatMessage, User: ChatUser>: ViewModifi
...
                     .shadow(
-                        color: imageStyle.shadowColor,
+                        color: show ? imageStyle.shadowColor: Color.clear,
                         radius: imageStyle.shadowRadius
                     )
             )
+
+    }
+    
+    private var novatar: some View {
```
Comment (gabrielPeart) [inline Sources/SwiftyChat/Common/Modifiers/AvatarModifier.swift:81]: noAvatar

File: Sources/SwiftyChat/Common/Modifiers/AvatarModifier.swift [hunk 4]
```diff
@@ -31,6 +31,8 @@ internal struct AvatarModifier<Message: ChatMessage, User: ChatUser>: ViewModifi
     /// Current avatar position
     private var currentAvatarPosition: AvatarPosition { isSender ? outgoingAvatarPosition : incomingAvatarPosition }
     
+    public let show: Bool
```
Comment (gabrielPeart) [inline Sources/SwiftyChat/Common/Modifiers/AvatarModifier.swift:34]: Could we rename this property to give more context regarding the purpose?

**Review by gabrielPeart — COMMENTED** [review]
Nice stuff! Thanks for contributing.
I have recommended some possible enhancements

**Comment by EnesKaraosman** [comment]
@blakesplay can you update PR based on review comments

**Comment by blakesplay** [comment]
Will do today or tomorrow, have been busy lately.

File: Sources/SwiftyChat/Common/Modifiers/AvatarModifier.swift (hunk 3 above)
Comment (blakesplay) [reply Sources/SwiftyChat/Common/Modifiers/AvatarModifier.swift:81]: I named it blankAvatar for extra clarity.

File: Sources/SwiftyChat/Common/Modifiers/AvatarModifier.swift (hunk 4 above)
Comment (blakesplay) [reply Sources/SwiftyChat/Common/Modifiers/AvatarModifier.swift:34]: Renamed to showAvatarForMessage

File: Sources/SwiftyChat/ChatView.swift (hunk 2 above)
Comment (blakesplay) [reply Sources/SwiftyChat/ChatView.swift:166]: Good idea, done.

File: Sources/SwiftyChat/ChatView.swift (hunk 1 above)
Comment (blakesplay) [reply Sources/SwiftyChat/ChatView.swift:124]: New helper function called shouldShowAvatarForMessage

**Review by gabrielPeart — APPROVED** [review]

**Pull request merged**